web/libs/editor/src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx

```diff
@@ -206,37 +207,7 @@ const useDataTree = ({ regions, rootClass, footer }: any) => {
     const color = chroma(style ?? "#666").alpha(1);
     const mods: Record<string, any> = { hidden, type, isDrawing };
 
-    const label = (() => {
-      if (!type) {
-        return "No Label";
-      }
-      if (type.includes("label")) {
-        return item.value;
-      }
-      if (type.includes("region") || type.includes("range")) {
-        const labelsInResults = item.labelings.map((result: any) => result.selectedLabels || []);
-
-        const labels: any[] = [].concat(...labelsInResults);
-
-        return (
-          <Block name="labels-list">
-            {labels.map((label, index) => {
-              const color = label.background || "#000000";
-
-              return [
-                index ? ", " : null,
-                <Elem key={label.id} style={{ color }}>
-                  {label.value || "No label"}
-                </Elem>,
-              ];
-            })}
-          </Block>
-        );
-      }
-      if (type.includes("tool")) {
-        return item.value;
-      }
-    })();
+    const label = <RegionLabel item={item} />;
```

Comment:
```
    const label = <RegionLabel item={item} />;
```
❤️

Reply:
That was Andrew's Idea :) https://github.com/HumanSignal/label-studio/pull/6400#discussion_r1765078142